texar/torch/modules/connectors/connectors.py

```diff
@@ -128,12 +128,8 @@ def _mlp_transform(inputs: TensorStruct,
     """
     # Flatten inputs
     flat_input = nest.flatten(inputs)
-    if len(flat_input[0].size()) == 1:
-        batch_size = 1
-    else:
-        batch_size = flat_input[0].size(0)
-    flat_input = [x.view(-1, x.size(-1)) for x in flat_input]
-    concat_input = torch.cat(flat_input, 0)
+    flat_input = [x.view(-1, np.prod(list(x.size())[1:])) for x in flat_input]
```

Comment:
I think the original implementation is correct here, since linear layers work on the final dimension.